srml/bridge/src/lib.rs

```diff
@@ -49,36 +58,34 @@ use system::{ensure_signed};
 #[derive(Encode, Decode, Clone, PartialEq)]
 #[cfg_attr(feature = "std", derive(Debug))]
 pub struct BridgeInfo<T: Trait> {
-	last_finalized_block_number: T::BlockNumber,
-	last_finalized_block_hash: T::Hash,
-	last_finalized_state_root: T::Hash,
-	current_validator_set: Vec<(AuthorityId, AuthorityWeight)>,
+	last_finalized_block_header: T::Header,
+	current_validator_set: AuthorityList,
 }
 
 impl<T: Trait> BridgeInfo<T> {
 	pub fn new(
-			block_number: &T::BlockNumber,
-			block_hash: &T::Hash,
-			state_root: &T::Hash,
-			validator_set: Vec<(AuthorityId, AuthorityWeight)>,
+			block_header: T::Header,
+			validator_set: AuthorityList,
 		) -> Self
 	{
-		// I don't like how this is done, should come back to...
 		BridgeInfo {
-			last_finalized_block_number: *block_number,
-			last_finalized_block_hash: *block_hash,
-			last_finalized_state_root: *state_root,
+			last_finalized_block_header: block_header,
 			current_validator_set: validator_set,
 		}
 	}
 }
 
 type BridgeId = u64;
 
-pub trait Trait: system::Trait {}
+pub trait Trait: system::Trait<Hash=H256> {
+	type Block: BlockT<Hash=H256, Header=Self::Header>;
```

Comment:
nit: I think you can drop the `Hash=H256`, as it is implied by the `Header=Self::Header`.

Reply:
Looks like the compiler isn't happy without the `Hash=H256`.